### ISSUE TITLE
Specific version of solcjs and go-ethereum occurs error while calling smart contract in Hyperledger Besu

### ISSUE THREAD

**Issue by wnjoon** [hyperledger/besu#6175]
### Description

I am developing EVM-related client using Golang and found something error situation. 

- Tested smart contract : Openzepellin ERC20
- Solcjs version : 0.8.23 (latest)
- go-ethereum version : 1.13.x (latest)

FIrst, I compiled smart contract using solejs and send transaction using ABI and bytecode, not using go file generated by abigen, and it results 'transaction reverted'. 

So I downgraded solcjs version to 0.8.19 (less than 0.8.20), and it worked.
However, when I upgrade go-ethereum version to 1.13, still getting error.

In conclusion, When I tested several times, To use go-ethereum for developing Hyperledger Besu-related client needs
- go-ethereum version : 1.12 >=
- solcjs version : 0.8.19 >=

### Versions (Add all that apply)
* Software version: v23.10.1
* Go version: 1.21.3
* OS Name & Version: MacOS Sonoma 14.0

**Comment by non-fungible-nelson** [comment]
Hi there - are you running a private network with a combination of Geth and Besu nodes? Did you specify in your Besu genesis file the correct Fork milestones? It is possible your Besu node is attempting to use an older fork or milestone that doesn't support certain opcodes you used / deployed with Geth latest. [Check out the documentation here](https://besu.hyperledger.org/development/public-networks/reference/genesis-items#milestone-blocks) and see if it can be resolved

Any other information you can provide would be helpful! Thank you.
